plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -107,29 +113,42 @@ public long getSize()
         return columnSizes;
     }
 
-    public static class IcebergStatisticsBuilder
+    public static class Builder
     {
         private final Map<Integer, Type.PrimitiveType> idToTypeMapping;
         private final List<Types.NestedField> columns;
-        private final TypeManager typeManager;
         private final Map<Integer, Optional<Long>> nullCounts = new HashMap<>();
         private final Map<Integer, ColumnStatistics> columnStatistics = new HashMap<>();
         private final Map<Integer, Long> columnSizes = new HashMap<>();
+        private final Map<Type, io.trino.spi.type.Type> icebergToTrinoType;
+        private final Map<io.trino.spi.type.Type, MethodHandle> typeToComparisonHandle;
+
         private long recordCount;
         private long fileCount;
         private long size;
 
-        public IcebergStatisticsBuilder(
+        public Builder(
                 Map<Integer, Type.PrimitiveType> idToTypeMapping,
                 List<Types.NestedField> columns,
                 TypeManager typeManager)
         {
-            this.idToTypeMapping = requireNonNull(idToTypeMapping, "idToTypeMapping is null");
-            this.columns = requireNonNull(columns, "columns is null");
-            this.typeManager = requireNonNull(typeManager, "typeManager is null");
+            this.idToTypeMapping = ImmutableMap.copyOf(requireNonNull(idToTypeMapping, "idToTypeMapping is null"));
+            this.columns = ImmutableList.copyOf(requireNonNull(columns, "columns is null"));
+
+            this.icebergToTrinoType = columns.stream()
+                    .map(Types.NestedField::type)
+                    .distinct()
+                    .collect(toImmutableMap(Function.identity(), icebergType -> toTrinoType(icebergType, typeManager)));
+
+            this.typeToComparisonHandle = icebergToTrinoType.values().stream()
```

Comment:
Mapping them here is useful when you have many columns of same type, with stats.
And it only matters when # files isn't huge. 

Doing this in `ColumnStatistics::new` is simpler and IMO sufficient, as you do lookup per column.
(originally you had lookup per column x file)